.github/workflows/pulsar-ci-flaky.yaml

```diff
@@ -22,9 +22,27 @@ on:
   pull_request:
     branches:
       - master
+    paths-ignore:
+      # Doc
+      - 'site2/**'
+      - 'deployment/**'
+      - '.asf.yaml'
+      - '*.md'
+      - '**/*.md'
+      # CPP
+      - 'pulsar-client-cpp/**'
   push:
     branches:
       - branch-*
+    paths-ignore:
+      # Doc
+      - 'site2/**'
+      - 'deployment/**'
+      - '.asf.yaml'
+      - '*.md'
+      - '**/*.md'
+      # CPP
+      - 'pulsar-client-cpp/**'
```

Comment:
there might be some consequences in this change because paths-ignore works in a certain way. I'm not exactly sure if this would be a good thing to do.

Reply:
the [doc](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions) says 
>When all the path names match patterns in paths-ignore, the workflow will not run. If any path names do not match patterns in paths-ignore, even if some path names match the patterns, the workflow will run.

what kind of risks you think we run ?

Reply:
As long as all flaky tests are not cpp/py tests, this changes should be good.

Reply:
@tisonkun flaky tests suite contains only java tests. 
btw @lhotari I moved back to the changed_files check

Reply:
For me, [paths-filter](https://github.com/apache/pulsar-test-infra/tree/master/paths-filter) is there because currently GitHub doesn't support `paths` and `paths-ignore` in job level or step level. So it should be effectively the same as those two filters.

I don't object to use a `changed_files_job`. Thanks for your effort.